<!--
ナビゲーションのデモ
2015.02.18 Okada,H.
-->
<launch>
  <param name="use_sim_time" value="false" />

  <!-- Map server -->
  <arg name="map" default="map130.yaml" />
  <!-- Run the map server with the desired map -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(find turtlebot9_navigation)/maps/$(arg map)"/>


  <!-- Start the move_base node with the nav_test parameters -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <rosparam file="$(find turtlebot9_navigation)/config/turtlebot/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find turtlebot9_navigation)/config/turtlebot/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find turtlebot9_navigation)/config/turtlebot/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find turtlebot9_navigation)/config/turtlebot/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find turtlebot9_navigation)/config/turtlebot/base_local_planner_params.yaml" command="load" />
    <rosparam file="$(find turtlebot9_navigation)/config/nav_test_params.yaml" command="load" />
  </node>

 
  <arg name="initial_pose_x" default="2.0"/> 
  <arg name="initial_pose_y" default="1.0"/>
  <arg name="initial_pose_a" default="0.0"/>
  <include file="$(find turtlebot9_navigation)/launch/includes/amcl.launch.xml">
    <arg name="initial_pose_x" value="$(arg initial_pose_x)"/>
    <arg name="initial_pose_y" value="$(arg initial_pose_y)"/>
    <arg name="initial_pose_a" value="$(arg initial_pose_a)"/>
  </include>

  <!-- Start the navigation test -->
  <node pkg="turtlebot9_navigation" type="nav_test1.py" name="nav_test" output="screen">
    <param name="rest_time" value="10" />
    <param name="fake_test" value="false" />
  </node>

</launch>
